helpers/sendgrid.request.js

```diff
@@ -0,0 +1,42 @@
+require('dotenv').config({ path: './config/.env' });
+const constants = require('../config/constants.json');
+const _ = require('underscore');
+
+async function getSendgridMessages (startDate, endDate, emails) {
+  const request = require('request-promise');
+
+  const options = {
+      url: `${constants.sendgridapi}/messages`,
+      qs: {
+        query: `(Contains(events,"processed")) AND last_event_time BETWEEN TIMESTAMP "${startDate}" AND TIMESTAMP "${endDate}" AND (${emails})`,
+        limit: constants.sendgridResponseLimit
+      },
+      headers: {
+        'User-Agent': 'Request-Promise',
+        'Authorization': 'Bearer ' + process.env.SENDGRIDACCESSTOKEN
+      },
+  };
+  return await request.get(options);
+}
+
+function buildEmailsForRequestQuery (emails) {
+  let emailsQuery = '';
+
+  emails.forEach((email, index)=> {
+    emailsQuery = emailsQuery + `to_email="${email}" OR `;
+    if (index === emails.length - 1) {
+      emailsQuery = emailsQuery + `to_email="${email}"`;
+    }
+  });
+  return emailsQuery;
+}
+
+async function getEmailsWithActivityOnSendgrid (startDate, endDate, emails) {
+  const emailsQuery = buildEmailsForRequestQuery(emails);
+  const responseSendgrid = await getSendgridMessages(startDate, endDate, emailsQuery);
+  const parseSendgridResponse = JSON.parse(responseSendgrid).messages;
+  const processedEmailAddresses = _.union(_.pluck(parseSendgridResponse, 'to_email'));
```

Comment:
```suggestion
  const processedEmailAddresses = parseSendgridResponse.map(elem => elem.to_email);
```

This way is probably easier to read, what do you think?